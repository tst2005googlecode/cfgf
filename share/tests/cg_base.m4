m4_define(`foo', `bar')m4_dnl

# cg_if
cg_if(`2 + 2 == 4',
       	`indeed', `hey, this is a parallel universe !')
cg_if(`0x2B | !0x2B == 42',
       	`sure', `it might be the question but is not the answer')

# cg_resyscmd
cg_resyscmd(`echo "foo bar baz" | tr " " "\n"', `^ba(.)')

# cg_quote
cg_quote(0, `foo')
cg_quote(1, `foo')
cg_quote(2, `foo')
cg_quote(3, `foo')
cg_quote(4, `foo')
cg_quote(0, `foo', `foo')
cg_quote(1, `foo', `foo')
cg_quote(2, `foo', `foo')
cg_quote(3, `foo', `foo')
cg_quote(4, `foo', `foo')

