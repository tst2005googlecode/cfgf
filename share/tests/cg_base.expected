
# cg_if
indeed
it might be the question but is not the answer

# cg_resyscmd
r,
z

# cg_quote
bar
foo
`foo'
``foo''
```foo'''
bar,bar
foo,foo
`foo',`foo'
``foo',`foo''
```foo',`foo'''

